### PR TITLE
Geospatial techno

### PR DESCRIPTION
**Pull request by geospatialtechno**
We'll discuss two common methods for installing GeoServer software on Windows

### CODE REVIEW

File: _posts/2023-11-23-geospatial-techno.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+| [Facebook](https://www.facebook.com/geospatialtechno)
+| [Reddit](https://www.reddit.com/user/geospatialtechno)
+| [X](https://twitter.com/geospatialtechn)
+)
+
+----
+
+### GeoServer installation methods: “Windows Installer” and “Web Archive”
+GeoServer installation methods: “Windows Installer” and “Web Archive”
+In this session, we will talk about how to install GeoServer software by two commonly methods in Windows. If you want to access the complete tutorial, simply click on the [link](https://youtu.be/G47vV2od4yc).
```
Comment (bradh) [inline _posts/2023-11-23-geospatial-techno.md:23]:
```suggestion
In this session, we will talk about how to install GeoServer software by two common methods in Windows. If you want to access the complete tutorial, simply click on the [link](https://youtu.be/G47vV2od4yc).
```

File: _posts/2023-11-23-geospatial-techno.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+There are many ways to install GeoServer on your system. This tutorial will use the Oracle JRE8 also known as JRE 1.8 and discuss the two most commonly used installation methods for windows.
+- Windows Installer
+- Web Archive
+
+### Windows installer
+The Windows installer provides an easy way to set up GeoServer on your system, as it requires no configuration files to be edited or command line settings.
+
+#### Installation
+- Navigate to Oracle.com and download JRE 8 from the Products tab. Consider the operating system architecture and memory requirements when selecting a JRE installer. 32-bit Java version is restricted to 2 GB memory, while the 64-bit version is recommended for optimal server memory. Utilizing JAI with the 32-bit JRE can enhance performance for WMS output generation and raster operations.
```
Comment (bradh) [inline _posts/2023-11-23-geospatial-techno.md:44]: Why JRE 8?

File: _posts/2023-11-23-geospatial-techno.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+- Navigate to the GeoServer.org and Download page. Select Web Archive on the download page from the version of GeoServer that you wish to download.
+- Deploy the geoserver web archive as you would normally. Often, all that is necessary is to copy the geoserver.war file to the Tomcat’s webapps directory, then the application will be deployed automatically.
+- Now to access the Web administration interface, open a browser and navigate to localhost:8080 and press Manager App button. Enter the username and password of apache tomcat. Click on the start button for the geoserver. Once it has started, click the geoserver link. This will take you to the geoserver web page.
+
+#### Uninstallation
+Stop the container application. Remove the GeoServer webapp from the container application’s webapps directory. This will usually include the geoserver.war file as well as a geoserver directory.
+
+### Difference between GEOSERVER.war and GEOSERVER.exe?
+- geoserver.exe installs GeoServer as a windows service running inside Jetty. geoserver.war is a platform independant that needs to be installed into a container server e.g. tomcat, jboss, …
```
Comment (bradh) [inline _posts/2023-11-23-geospatial-techno.md:73]:
```suggestion
- geoserver.exe installs GeoServer as a windows service running inside Jetty. geoserver.war is a platform independent package that needs to be installed into a container server e.g. tomcat, jboss, …
```

**Review by bradh — CHANGES_REQUESTED** [review]
Could probably use more proof-reading / editing.

File: _posts/2023-11-23-geospatial-techno.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+[![](https://img.youtube.com/vi/G47vV2od4yc/0.jpg)](https://www.youtube.com/watch?v=G47vV2od4yc)
+
+### Introduction
+GeoServer can be installed on different operating systems, since it's a Java based application. You can run it on any kind of operating system for which exists a Java virtual machine. GeoServer’s speed depends a lot on the chosen Java Runtime Environment (JRE). The latest versions of GeoServer are tested with both OracleJRE and OpenJDK. These versions are:
+- Java 17 for GeoServer 2.23 and above
+- Java 11 for GeoServer 2.15 and above
+- Java 8 for GeoServer 2.9 to GeoServer 2.22
+- Java 7 for GeoServer 2.6 to GeoServer 2.8
+- Java 6 for GeoServer 2.3 to GeoServer 2.5
+- Java 5 for GeoServer 2.2 and earlier
```
Comment (jodygarnett) [inline _posts/2023-11-23-geospatial-techno.md:34]: It is not so useful to talk about old versions; we should stick with the currently supported versions only.
While some commercial providers can offer to support older environments - our community volunteers are already stretched to support a stable and maintainance branch :)

File: _posts/2023-11-23-geospatial-techno.md (hunk 2 above)
Comment (jodygarnett) [reply _posts/2023-11-23-geospatial-techno.md:44]: Actually we cannot recommend oracle JDK at this time either.

I expect this blog post would do better to link to the online instructions - https://docs.geoserver.org/latest/en/user/installation/index.html#installation ?

File: _posts/2023-11-23-geospatial-techno.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+- Review the selections, install GeoServer, and start it either manually or as a service.
+- Finally, navigate to localhost:8080/geoserver (or wherever you installed GeoServer) to access the GeoServer Web administration interface.
+
+#### Uninstallation
+GeoServer can be uninstalled in two ways:
+- By running the uninstall.exe file in the directory where GeoServer was installed
+- By standard Windows program removal
+
+### Web archive
+GeoServer is packaged as a standalone servlet for use with existing application servers such as Apache Tomcat, GlassFish, Jetty and JBoss. It has been mostly tested using Tomcat, and so is the recommended application server. There are reasons for installing it such as it is widely used, well-documented, and relatively simple to configure. GeoServer requires a newer version of Tomcat (7.0.65 or later) that implements Servlet 3 and annotation processing. Other application servers have been known to work, but are not guaranteed.
```
Comment (jodygarnett) [inline _posts/2023-11-23-geospatial-techno.md:59]: This sentence does not make sense to me:

1. It is packaged as a web-archive
2. It is not standalone anything; as a web-archive requires an application server to run :)

```
GeoServer is packaged as a web-archive (WAR) for use with an application server such as Apache Tomcat or Jetty.
```

Glassfish and JBoss are not used so much anymore.

Comment (geospatialtechno) [reply _posts/2023-11-23-geospatial-techno.md:59]: You are right

File: _posts/2023-11-23-geospatial-techno.md (hunk 3 above)
Comment (geospatialtechno) [reply _posts/2023-11-23-geospatial-techno.md:73]: You are right

File: _posts/2023-11-23-geospatial-techno.md (hunk 2 above)
Comment (geospatialtechno) [reply _posts/2023-11-23-geospatial-techno.md:44]: You are right

File: _posts/2023-11-23-geospatial-techno.md (hunk 1 above)
Comment (geospatialtechno) [reply _posts/2023-11-23-geospatial-techno.md:23]: You are right

File: _posts/2023-11-23-geospatial-techno.md (hunk 4 above)
Comment (aaime) [reply _posts/2023-11-23-geospatial-techno.md:34]: You're both right. But users need to be aware that older versions are unsupported and won't receive fixes nor security updates, and contain well-known security vulnerabilities that have not been patched (so use at own risk).

**Comment by geospatialtechno** [comment]
Thank you all so much. I'll definitely consider the points you mentioned and make the necessary edits.

File: _posts/2023-11-23-geospatial-techno.md (hunk 2 above)
Comment (jodygarnett) [reply _posts/2023-11-23-geospatial-techno.md:44]: Nope- oracle does not offer a Java 11 anymore for free (it is too old and they do not official oracle Java anymore for production use unless you pay). 

We recommend OoenJDK (included with your Linux distribution) or from Adoptium 

Please review the current install instructions and update your blog post.

**Comment by jodygarnett** [comment]
Has all changes been addressed now?

File: _posts/2023-11-23-geospatial-techno.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+- Navigate to the GeoServer.org and Download page. Select Web Archive on the download page from the version of GeoServer that you wish to download.
+- Deploy the GeoServer web archive as you would normally. Often, all that is necessary is to copy the GeoServer.war file to the Tomcat’s webapps directory, then the application will be deployed automatically.
+- Now to access the Web administration interface, open a browser and navigate to localhost:8080 and press Manager App button. Enter the username and password of apache tomcat. Click on the start button for the GeoServer. Once it has started, click the GeoServer link. This will take you to the GeoServer web page.
+
+#### Uninstallation
+Stop the container application. Remove the GeoServer webapp from the container application’s webapps directory. This will usually include the GeoServer.war file as well as a GeoServer directory.
+
+### Difference between GEOSERVER.war and GEOSERVER.exe?
+- GeoServer.exe installs GeoServer as a windows service running inside Jetty. GeoServer.war is a platform independant package that needs to be installed into a container server e.g. tomcat, jboss, …
```
Comment (jodygarnett) [inline _posts/2023-11-23-geospatial-techno.md:78]:
```suggestion
- The `GeoServer.exe` NSIS installer registers GeoServer as a Windows Service, which uses the Jetty application server to run GeoServer. The `geoserver.war`  is a platform independent web-archive package to be deployed in your own application server (we recommend Apache Tomcat.) 
```

File: _posts/2023-11-23-geospatial-techno.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+The downside of using the Windows Installer is that you have to be on a Windows machine, you also get less choice of servlet container.
```
Comment (jodygarnett) [inline _posts/2023-11-23-geospatial-techno.md:79]: This sentence does not make sense to me? This page is dedicated to installing on windows so being on a windows machine is not a downside?

```
- Using the `geoserver.exe` installer is a reliable way to setup GeoServer as a windows background service. The downside is the included Jetty application server is managed using text files (jetty.ini) once installed.
```

File: _posts/2023-11-23-geospatial-techno.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+- Sometimes you need to install multiple GeoServers on your computer at the same time. In Windows mode, you have to assign different ports to each GeoServer and install them, while in a Web Archive,
```
Comment (jodygarnett) [inline _posts/2023-11-23-geospatial-techno.md:80]: Is this required or recommend anymore? Now that marline provides good multi-core performance I cannot see why I would run several geoserver at the same time.

```
- Use of `geoserver.war` web-archive is provided to install into your own application server (we recommend Apache Tomcat as the market leader, with excellent documentation and integration options). A single application server may support several web application allowing GeoServer to be run alongside your own java web application.
```

File: _posts/2023-11-23-geospatial-techno.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+#### Uninstallation
+Stop the container application. Remove the GeoServer webapp from the container application’s webapps directory. This will usually include the GeoServer.war file as well as a GeoServer directory.
+
+### Difference between GEOSERVER.war and GEOSERVER.exe?
+- GeoServer.exe installs GeoServer as a windows service running inside Jetty. GeoServer.war is a platform independant package that needs to be installed into a container server e.g. tomcat, jboss, …
+The downside of using the Windows Installer is that you have to be on a Windows machine, you also get less choice of servlet container.
+- Sometimes you need to install multiple GeoServers on your computer at the same time. In Windows mode, you have to assign different ports to each GeoServer and install them, while in a Web Archive, 
+you can easily install many GeoServers on one port with different names in much less time.
```
Comment (jodygarnett) [inline _posts/2023-11-23-geospatial-techno.md:81]: I still do not understand why this is of value?

File: _posts/2023-11-23-geospatial-techno.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+There are many ways to install GeoServer on your system. This tutorial will cover the two most commonly used installation methods on Windows.
+- Windows Installer
+- Web Archive
+
+### Windows installer
+The Windows installer provides an easy way to set up GeoServer on your system, as it requires no configuration files to be edited or command line settings.
+
+#### Installation
+- GeoServer requires a Java environment (JRE) to be installed on your system, available from [OpenJDK](https://openjdk.org) or [Adoptium](https://adoptium.net) for Windows installer, or provided by your OS distribution.
```
[the remark this answers is not in the document]
Comment (geospatialtechno) [reply _posts/2023-11-23-geospatial-techno.md:46]: > I would just recommend Adoption here as it has a very good MSI installer and it is what we recommend on windows.
> 
> If you follow the link to the OpenJDK website they give you instructions to download https://jdk.java.net/21/ etc...

Sure, I'll edit it.

File: _posts/2023-11-23-geospatial-techno.md (hunk 6 above)
Comment (geospatialtechno) [reply _posts/2023-11-23-geospatial-techno.md:78]: > 

You are right

**Comment by geospatialtechno** [comment]
@jodygarnett I have made the corrections you suggested and recommitted again.

**Pull request merged**